pkg/http/server.go

```diff
@@ -259,47 +281,39 @@ func (s *server) usePublicRouter() fasthttp.RequestHandler {
 	return router.Handler
 }
 
-func (s *server) useComponents(next fasthttp.RequestHandler) fasthttp.RequestHandler {
-	return fasthttpadaptor.NewFastHTTPHandler(
-		s.pipeline.Apply(
-			nethttpadaptor.NewNetHTTPHandlerFunc(next),
-		),
-	)
+func (s *server) useComponents(next http.Handler) http.Handler {
+	return s.pipeline.Apply(next)
 }
 
-func (s *server) useCors(next fasthttp.RequestHandler) fasthttp.RequestHandler {
+func (s *server) useCors(next http.Handler) http.Handler {
+	// TODO: Technically, if "AllowedOrigins" is "*", all origins should be allowed
```

Comment:
Reviewers: Dapr is likely behaving incorrectly here. Setting "*" in CORS should mean "allow all origins", but Dapr <=1.10 was actually omitting the CORS headers entirely. Since this is a backwards-incompatible change, I am not touching it now.